atlas/schema.hcl

```diff
@@ -1,11 +1,492 @@
+    on_delete   = CASCADE
+  }
+index "project_id_idx" {
+    unique  = true
+    columns = [column.project_id]
+  }
+}
+table "scan_inference_v1" {
+  schema = schema.public
+  comment = "This table stores the results of inferencing on the image scans."
+  column "scan_id" {
+    null = false
+    type = bigint
+  }
+  column "scan_inference_id" {
+    null = false
+    type = bigint
+    identity {
+      generated = ALWAYS
+      start = 0
+      increment = 1
+    }
+  }
+  column "inference_result" {
+    null = false
+    type = jsonb
+    comment = "JSON of image scan inference output."
   }
   primary_key {
-    columns = [column.id]
+    columns = [column.scan_id, column.scan_inference_id]
+  }
+  foreign_key "scan_id" {
+    columns     = [column.scan_id]
+    ref_columns = [table.scan_v1.column.scan_id]
+    on_update   = CASCADE
+    on_delete   = CASCADE
+  }
+}
+table "scan_v1" {
+  schema = schema.public
+  comment = "Stores the compressed images to be scanned and inferenced."
+  column "project_id" {
+    null = false
+    type = bigint
+  }
+  column "scan_id" {
+    null = false
+    type = bigint
+    identity {
+      generated = ALWAYS
+      start = 0
+      increment = 1
+    }
+  }
+  column "image" {
+    null = false
+    type = bytea
+    comment = "The storage of the compressed image file."
+  }
+  column "create_time" {
+    null = false
+    type = timestamptz
+  }
+  primary_key {
+    columns = [column.project_id, column.scan_id]
+  }
+  foreign_key "project_id" {
+    columns     = [column.project_id]
+    ref_columns = [table.project_v1.column.project_id]
+    on_update   = CASCADE
+    on_delete   = CASCADE
+  }
+index "scan_id_idx" {
+    unique  = true
+    columns = [column.scan_id]
+  }
+}
+table "user_snapshot_v1" {
+  schema = schema.public
+  comment = "A snapshot in time of a user profile. A new snapshot is generated everytime a user modifies information related to their profile."
+  column "user_id" {
+    null = false
+    type = bigint
+  }
+  column "snapshot_id" {
+    null = false
+    type = bigint
+    identity {
+      generated = ALWAYS
+      start = 0
+      increment = 1
+    }
+  }
+  column "given_name" {
+    null = false
+    type = text
+  }
+  column "middle_name" {
+    null = false
+    type = text
+  }
+  column "family_name" {
+    null = false
+    type = text
+  }
+  column "date_of_birth" {
+    null = false
+    type = date
+  }
+  column "marital_status" {
+    null = false
+    type = text
+  }
+  column "citizenship" {
+    null = false
+    type = text
+  }
+  column "email" {
+    null = false
+    type = text
+  }
+  column "create_time" {
+    null = false
+    type = timestamptz
+  }
+  primary_key {
+    columns = [column.user_id, column.snapshot_id]
+  }
+  foreign_key "user_id" {
+    columns     = [column.user_id]
+    ref_columns = [table.user_v1.column.user_id]
+    on_update   = CASCADE
+    on_delete   = CASCADE
+  }
+}
+table "user_v1" {
```

Comment:
again why is there a duplicate of fields here, when fields are already tracked in the user snapshot table?